### PR TITLE
Decode morse

### PR DESCRIPTION
**Pull request by RileyManda**
- [x] Added readme
- [x] Project setup and initialization
- [x] Updated folder name
- [x] Linter fixes
-  Create a method to decode the entire message in Morse code
-   Create a method to decode a Morse code character
- Create a method to decode the entire word

### CODE REVIEW

File: decoding_morse.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+def decode_message(morse_message)
+  words = morse_message.split('   ')
+  decoded_words = words.map do |_word|
+    characters = word.chars
+    decoded_characters = characters.map { |char| decode_char(char) }
+    decoded_characters.join
+  end
+
+  decoded_words.join('   ')
+end
```
Comment (V-Blaze) [inline decoding_morse.rb]: - You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :smile: :wave: 

- [ ] According to the project requirement, you are meant to run this test to see your decode morse code in action.

```sh
p decode_char('.-')
p decode_word('-- -.--')
p decode_message("-- -.--   -. .- -- .")
p decode_message(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")
```
-  [ ] When I ran these, I got an error in the console, I'll suggest you go through your methods and make sure everything is implemented correctly, and then run the code to make sure the result you get is what's expected. :smile: :small_airplane: :wave: 

- - Screenshot for reference:
Screenshot from 2023-08-30 11-59-13

File: README.md [hunk 2]
```diff
@@ -1 +1,244 @@
-# morse-code
\ No newline at end of file
+<a name="readme-top"></a>
+
+<div align="center">
+
+  <img src="app-logo.png" alt="logo" width="140"  height="auto" />
+  <br/>
```
Comment (V-Blaze) [inline README.md]: - [ ] You are meant to remove this Microverse logo as it is a placeholder that came with the template. You can add your logo if you have any or leave out the logo entirely. :small_airplane: :wave: 

- - Screenshot:
Screenshot from 2023-08-30 12-03-10

File: README.md [hunk 3]
```diff
@@ -1 +1,244 @@
...
+- This project is based on the Original design by: [Nelson Sakwa
+](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))
+- Original logo design from: [Svgrepo](https://www.svgrepo.com/)
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+[![MIT License](https://img.shields.io/badge/License-MIT-green.svg)](./LICENSE)
+
```
Comment (V-Blaze) [inline README.md]: - [ ] The License link you have in your readme leads to a 404 error page. I'll suggest you add a Licence file and correctly link it in your readme. :smile: :wave: 

- - Screenshot:
Screenshot from 2023-08-30 12-05-45

File: README.md [hunk 4]
```diff
@@ -1 +1,244 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ATTRIBUTION -->
+
+## 👥 Attribution <a name="attribution"></a>
+- This project is based on the Original design by: [Nelson Sakwa
+](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))
+- Original logo design from: [Svgrepo](https://www.svgrepo.com/)
```
Comment (V-Blaze) [inline README.md]: - [ ] This project's design is not from `Nelson Sakwa`, I'll suggest you update your readme with accurate information about the project as a good-looking readme makes your project look professional. :smile: :small_airplane:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @RileyManda, @ClarenceAtim  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Well-documented code✔
-  Correct Git flow👋 
-  Linters setup correctly ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RileyManda** [comment]
We have completed making the requested changes from the previous reviewer:

-Ensured the following commands are executing successfully on the terminal:
p decode_char('.-')
p decode_word('-- -.--')
p decode_message("-- -.--   -. .- -- .")
p decode_message(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")


Screen Shot 2023-08-30 at 1 28 54 PM

-Updated readme file and remove attributions  and project setup
-Updated readme file and remove logo

File: README.md (hunk 4 above)
Comment (RileyManda) [reply README.md]: Thank you for the review @V-Blaze 
The readme file template was copied from a previous project.
We will make the changes to submit for review.

File: README.md (hunk 2 above)
Comment (RileyManda) [reply README.md]: Noted with Thanks @V-Blaze

File: decoding_morse.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+def decode_message(morse_message)
+  words = morse_message.split('   ')
+  decoded_words = words.map do |word|
+    characters = word.chars
+    decoded_characters = characters.map { |char| decode_char(char) }
+    decoded_characters.join
+  end
+
+  decoded_words.join('   ')
+end
```
Comment (LionRouge1) [inline decoding_morse.rb]: - [ ] Good job creating the `decode_char` and the `decode_word` separately :clap: However the `decode_message` function is not returning the correct message.
  This `decode_message("-- -.--   -. .- -- .")`  supposed to returns **`"MY NAME"`** but we got this **`"TTUNKNOWNTETT   TEUNKNOWNETUNKNOWNTTUNKNOWNE"`**
  Kindly make sure the correct message is returned by the `decode_message` function.

  **HIT:  We can simply use the `decode_word` function in our `decode_message` function. So the `decode_message` could looks this :point_down:** (Your code doesn't need to be exactly like this)

```ruby
def decode_message(morse_message)
  morse_message.split('   ').map { |w| decode_word(w) }.join(' ')
end
```

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @RileyManda && @ClarenceAtim, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

## Suggested changes :recycle: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RileyManda** [comment]
Screen Shot 2023-08-30 at 2 45 38 PM

Screen Shot 2023-08-30 at 3 19 18 PM

**Review by joel-kalema — APPROVED** [review]
Hi @RileyManda 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**